controllers/noderefutil/util.go

```diff
@@ -70,3 +70,17 @@ func IsNodeReady(node *corev1.Node) bool {
 	}
 	return false
 }
+
+// IsNodeUnreachable returns true if a node is unreachable.
+// Node is considered unreachable when its ready status is "Unknown".
+func IsNodeUnreachable(node *corev1.Node) bool {
+	if node == nil {
```

Comment:
I'm not sure we need this check. It should only ever be called with a valid node.